api4/file.go

```diff
@@ -737,7 +737,7 @@ func getPublicFile(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func writeFileResponse(filename string, contentType string, contentSize int64, webserverMode string, fileReader io.Reader, forceDownload bool, w http.ResponseWriter, r *http.Request) *model.AppError {
-	w.Header().Set("Cache-Control", "max-age=2592000, private")
+	w.Header().Set("Cache-Control", "no-cache")
```

Comment:
I think we might still want `private`.

Reply:
@crspeller `private` and `no-cache` both are `Cacheability` directives, while `private` indicates that this may be cached for a single user in a private cache (no shared ones), no-cache forces a revalidation directly. So the `no-cache` is essentially a stricter directive than private. See [1](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching) and [2](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#Cacheability)

Reply:
It's not strictly though. With no `no-cache` an intermediate proxy can store the response, but not return it without re-validating. With `private` the response "must not be stored by a shared cache" (from 1). You can also see that in the flowchart in 2. Essentially the difference is we are asking it not to even store the file (since it's private).
Also from experiance I know of at least one proxy that asks for forgiveness rather than permission on `no-cache`. (it sends the response then re-validates its cache)

Reply:
Diving deeper again on the subject, you are absolutely right. Sorry for the confusion and spreading the wrong information, I'll update the PR.